src/tracing/json/json_tracing.cpp

```diff
@@ -295,11 +296,28 @@ void JsonBackend::TraceCounter(const char * label)
     OutputValue(value);
 }
 
-void JsonBackend::TraceMetric(const char * label, int32_t val)
+void JsonBackend::LogMetricEvent(const MetricEvent & event)
 {
     ::Json::Value value;
-    value["label"] = label;
-    value["value"] = val;
+
+    value["label"] = event.key();
```

Comment:
having the key as part of the object is inconsisted with all other methods. Please split.

Reply:
Note that the ones without a label/key above (Lookup and such) have no key whatsoever ... they are so singular that their existance implies the key.

The metric is more close to begin/end/instant/counter and should be formatted as such.

Reply:
I think in time we should move to have the macro split out label from data, since every macro that has a label until now does take it in the macro not within the object. Can be a followup.